Code/Framework/AzCore/Tests/Memory/LeakDetection.cpp

```diff
@@ -6,13 +6,47 @@
  *
  */
 
-#include <AzCore/Memory/BestFitExternalMapAllocator.h>
+#include <AzCore/Memory/AllocationRecords.h>
+#include <AzCore/Memory/AllocatorManager.h>
+#include <AzCore/Memory/IAllocator.h>
 #include <AzCore/Memory/PoolAllocator.h>
 #include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/Memory/ChildAllocatorSchema.h>
 
 
 namespace UnitTest
 {
+    static void CheckAllocatorsForLeaks(bool leakExpected)
+    {
+        auto& allocatorManager = AZ::AllocatorManager::Instance();
+        const int numAllocators = allocatorManager.GetNumAllocators();
+        bool anyAllocatorHasAllocations = false;
+        allocatorManager.GarbageCollect();
+        for (int i = numAllocators - 1; i >= 0; --i)
+        {
+            auto* allocator = allocatorManager.GetAllocator(i);
+            const AZ::Debug::AllocationRecords* records = allocator->GetRecords();
+            if (records)
+            {
+                const bool hasAllocations = records->RequestedBytes() != 0;
+                anyAllocatorHasAllocations |= hasAllocations;
```

Comment:
This should use logical or.
```suggestion
                anyAllocatorHasAllocations = anyAllocatorHasAllocations || hasAllocations;
```

Reply:
Fixed.